src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -375,18 +375,33 @@ private SpaceType getSpaceType(final FieldInfo fieldInfo) {
 
     private boolean canDoExactSearch(final int filterIdsCount, final int searchableDocs) {
         log.debug(
-            "Info for doing exact search Live Docs: {}, filterIdsLength : {}, Threshold value: {} , Threshold %age : {}",
+            "Info for doing exact search Live Docs: {}, filterIdsLength : {}, Threshold value: {}",
             searchableDocs,
             filterIdsCount,
-            KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName()),
-            KNNSettings.getFilteredExactSearchThresholdPct(knnQuery.getIndexName())
+            KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName())
         );
+        int filterThresholdValue = KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName());
         // Refer this GitHub around more details https://github.com/opensearch-project/k-NN/issues/1049 on the logic
-        return filterIdsCount <= knnQuery.getK()
-            || (filterIdsCount <= KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName())
-                && (((float) filterIdsCount / (float) searchableDocs) * 100) <= (float) KNNSettings.getFilteredExactSearchThresholdPct(
-                    knnQuery.getIndexName()
-                ));
+        if (filterIdsCount <= knnQuery.getK()) {
+            return true;
+        }
+
+        if (isExactSearchThresholdSet(filterThresholdValue)
+            && KNNConstants.MAX_DISTANCE_COMPUTATIONS <= filterIdsCount * knnQuery.getQueryVector().length) {
```

Comment:
Better to just return `KNNConstants.MAX_DISTANCE_COMPUTATIONS <= filterIdsCount * knnQuery.getQueryVector().length` here because when threshold is not set, we know we will only use distance computation value to decide. 

```
if (isExactSearchThresholdSet(filterThresholdValue) == false) {
  return KNNConstants.MAX_DISTANCE_COMPUTATIONS <= filterIdsCount * knnQuery.getQueryVector().length;
}
```